sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/TypesafeBackendImpl.kt

```diff
@@ -27,29 +30,67 @@ internal class TypesafeBackendImpl(private val backend: Backend) : TypesafeBacke
     override val network: ZcashNetwork
         get() = ZcashNetwork.from(backend.networkId)
 
-    override suspend fun getAccounts(): List<Account> = backend.getAccounts().map { Account(it.accountIndex.toInt()) }
+    override suspend fun getAccounts(): List<Account> = backend.getAccounts().map { Account.new(it) }
 
     override suspend fun createAccountAndGetSpendingKey(
-        seed: ByteArray,
+        accountName: String,
+        keySource: String?,
+        seed: FirstClassByteArray,
         treeState: TreeState,
         recoverUntil: BlockHeight?
-    ): UnifiedSpendingKey {
-        return UnifiedSpendingKey(
+    ): AccountUsk {
+        return AccountUsk.new(
             backend.createAccount(
-                seed = seed,
+                accountName = accountName,
+                keySource = keySource,
+                seed = seed.byteArray,
                 treeState = treeState.encoded,
                 recoverUntil = recoverUntil?.value
             )
         )
     }
 
+    override suspend fun importAccountUfvk(
+        recoverUntil: BlockHeight?,
+        setup: AccountImportSetup,
+        treeState: TreeState,
+    ): Account {
+        return Account.new(
+            jniAccount =
+                when (setup.purpose) {
+                    is AccountPurpose.Spending ->
+                        backend.importAccountUfvk(
+                            accountName = setup.accountName,
+                            keySource = setup.keySource,
+                            purpose = setup.purpose.value,
+                            recoverUntil = recoverUntil?.value,
+                            treeState = treeState.encoded,
+                            ufvk = setup.ufvk.encoding,
+                            seedFingerprint = setup.purpose.seedFingerprint,
+                            zip32AccountIndex = setup.purpose.zip32AccountIndex?.index,
+                        )
+                    AccountPurpose.ViewOnly ->
+                        backend.importAccountUfvk(
+                            accountName = setup.accountName,
+                            keySource = setup.keySource,
+                            purpose = setup.purpose.value,
+                            recoverUntil = recoverUntil?.value,
+                            treeState = treeState.encoded,
+                            ufvk = setup.ufvk.encoding,
+                            seedFingerprint = null,
+                            zip32AccountIndex = null,
```

Comment:
Bug: these are not valid values for `seedFingerprint` and `zip32AccountIndex`.

Reply:
Discussion and findings moved to https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/issues/1651